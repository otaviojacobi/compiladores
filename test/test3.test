class Rectangle {

    private int width, height;

    static protected char* sample = "a test !";

    public int set_width(const int new_width) {
        width = new_width;
        return width;
    }

    public int get_width() {
        return width;
    }

    public int set_height(const int new_height) {
        height = new_height;
        return height;
    }

    public int get_height() {
        return height;
    }

    public int get_area() {
        return width * height;
    }

    public string get_sample_with_hello_msg(string hello_msg) {
        return to_string(sample) + hello_msg;
    }

    public int do_while_test() {

        int i = 0;

        do {
            putc(sample[i]);
            i++;
        } while (i < strlen(teste));

        return 0;
    }

    public int while_test() {

        int i = 0;

        while(i < strlen(teste)) {
            putc(sample[i]);
            i++;
        }

        return 0;
    }

    public int for_test() {

        for(int i = 0; i < strlen(teste); i++) {
            putc(sample[i]);
            i++;
        }

        return 0;
    }

    public int do_while_test_with_if() {

        int i = 0;

        do {
            if(i%2 == 0)
                putc(sample[i]);
            i++;
        } while (i < strlen(teste));

        return 0;
    }

    public int while_test_with_if() {

        int i = 0;

        while(i < strlen(teste)) {
            if(i <= 5) {
                putc(sample[i]);
            } else {
                putc(toupper(sample[i]))
            } 
            i++;
        }

        return 0;
    }

    public int for_test_with_if() {

        for(int i = 0; i < strlen(teste); i++) {
            if(i%3 == 0)
                continue;
            if(i%11 == 0)
                break;
            putc(sample[i]);
            i++;
        }

        return 0;
    }
}