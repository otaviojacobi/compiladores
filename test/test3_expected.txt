1 TK_PR_CLASS [class]
1 TK_IDENTIFICADOR [Rectangle]
1 TK_ESPECIAL [{]
3 TK_PR_PRIVATE [private]
3 TK_PR_INT [int]
3 TK_IDENTIFICADOR [width]
3 TK_ESPECIAL [,]
3 TK_IDENTIFICADOR [height]
3 TK_ESPECIAL [;]
5 TK_PR_STATIC [static]
5 TK_PR_PROTECTED [protected]
5 TK_PR_CHAR [char]
5 TK_ESPECIAL [*]
5 TK_IDENTIFICADOR [sample]
5 TK_ESPECIAL [=]
5 TK_LIT_STRING ["a test !"]
5 TK_ESPECIAL [;]
7 TK_PR_PUBLIC [public]
7 TK_PR_INT [int]
7 TK_IDENTIFICADOR [set_width]
7 TK_ESPECIAL [(]
7 TK_PR_CONST [const]
7 TK_PR_INT [int]
7 TK_IDENTIFICADOR [new_width]
7 TK_ESPECIAL [)]
7 TK_ESPECIAL [{]
8 TK_IDENTIFICADOR [width]
8 TK_ESPECIAL [=]
8 TK_IDENTIFICADOR [new_width]
8 TK_ESPECIAL [;]
9 TK_PR_RETURN [return]
9 TK_IDENTIFICADOR [width]
9 TK_ESPECIAL [;]
10 TK_ESPECIAL [}]
12 TK_PR_PUBLIC [public]
12 TK_PR_INT [int]
12 TK_IDENTIFICADOR [get_width]
12 TK_ESPECIAL [(]
12 TK_ESPECIAL [)]
12 TK_ESPECIAL [{]
13 TK_PR_RETURN [return]
13 TK_IDENTIFICADOR [width]
13 TK_ESPECIAL [;]
14 TK_ESPECIAL [}]
16 TK_PR_PUBLIC [public]
16 TK_PR_INT [int]
16 TK_IDENTIFICADOR [set_height]
16 TK_ESPECIAL [(]
16 TK_PR_CONST [const]
16 TK_PR_INT [int]
16 TK_IDENTIFICADOR [new_height]
16 TK_ESPECIAL [)]
16 TK_ESPECIAL [{]
17 TK_IDENTIFICADOR [height]
17 TK_ESPECIAL [=]
17 TK_IDENTIFICADOR [new_height]
17 TK_ESPECIAL [;]
18 TK_PR_RETURN [return]
18 TK_IDENTIFICADOR [height]
18 TK_ESPECIAL [;]
19 TK_ESPECIAL [}]
21 TK_PR_PUBLIC [public]
21 TK_PR_INT [int]
21 TK_IDENTIFICADOR [get_height]
21 TK_ESPECIAL [(]
21 TK_ESPECIAL [)]
21 TK_ESPECIAL [{]
22 TK_PR_RETURN [return]
22 TK_IDENTIFICADOR [height]
22 TK_ESPECIAL [;]
23 TK_ESPECIAL [}]
25 TK_PR_PUBLIC [public]
25 TK_PR_INT [int]
25 TK_IDENTIFICADOR [get_area]
25 TK_ESPECIAL [(]
25 TK_ESPECIAL [)]
25 TK_ESPECIAL [{]
26 TK_PR_RETURN [return]
26 TK_IDENTIFICADOR [width]
26 TK_ESPECIAL [*]
26 TK_IDENTIFICADOR [height]
26 TK_ESPECIAL [;]
27 TK_ESPECIAL [}]
29 TK_PR_PUBLIC [public]
29 TK_PR_STRING [string]
29 TK_IDENTIFICADOR [get_sample_with_hello_msg]
29 TK_ESPECIAL [(]
29 TK_PR_STRING [string]
29 TK_IDENTIFICADOR [hello_msg]
29 TK_ESPECIAL [)]
29 TK_ESPECIAL [{]
30 TK_PR_RETURN [return]
30 TK_IDENTIFICADOR [to_string]
30 TK_ESPECIAL [(]
30 TK_IDENTIFICADOR [sample]
30 TK_ESPECIAL [)]
30 TK_ESPECIAL [+]
30 TK_IDENTIFICADOR [hello_msg]
30 TK_ESPECIAL [;]
31 TK_ESPECIAL [}]
33 TK_PR_PUBLIC [public]
33 TK_PR_INT [int]
33 TK_IDENTIFICADOR [do_while_test]
33 TK_ESPECIAL [(]
33 TK_ESPECIAL [)]
33 TK_ESPECIAL [{]
35 TK_PR_INT [int]
35 TK_IDENTIFICADOR [i]
35 TK_ESPECIAL [=]
35 TK_LIT_INT [0]
35 TK_ESPECIAL [;]
37 TK_PR_DO [do]
37 TK_ESPECIAL [{]
38 TK_IDENTIFICADOR [putc]
38 TK_ESPECIAL [(]
38 TK_IDENTIFICADOR [sample]
38 TK_ESPECIAL [[]
38 TK_IDENTIFICADOR [i]
38 TK_ESPECIAL []]
38 TK_ESPECIAL [)]
38 TK_ESPECIAL [;]
39 TK_IDENTIFICADOR [i]
39 TK_ESPECIAL [+]
39 TK_ESPECIAL [+]
39 TK_ESPECIAL [;]
40 TK_ESPECIAL [}]
40 TK_PR_WHILE [while]
40 TK_ESPECIAL [(]
40 TK_IDENTIFICADOR [i]
40 TK_ESPECIAL [<]
40 TK_IDENTIFICADOR [strlen]
40 TK_ESPECIAL [(]
40 TK_IDENTIFICADOR [teste]
40 TK_ESPECIAL [)]
40 TK_ESPECIAL [)]
40 TK_ESPECIAL [;]
42 TK_PR_RETURN [return]
42 TK_LIT_INT [0]
42 TK_ESPECIAL [;]
43 TK_ESPECIAL [}]
45 TK_PR_PUBLIC [public]
45 TK_PR_INT [int]
45 TK_IDENTIFICADOR [while_test]
45 TK_ESPECIAL [(]
45 TK_ESPECIAL [)]
45 TK_ESPECIAL [{]
47 TK_PR_INT [int]
47 TK_IDENTIFICADOR [i]
47 TK_ESPECIAL [=]
47 TK_LIT_INT [0]
47 TK_ESPECIAL [;]
49 TK_PR_WHILE [while]
49 TK_ESPECIAL [(]
49 TK_IDENTIFICADOR [i]
49 TK_ESPECIAL [<]
49 TK_IDENTIFICADOR [strlen]
49 TK_ESPECIAL [(]
49 TK_IDENTIFICADOR [teste]
49 TK_ESPECIAL [)]
49 TK_ESPECIAL [)]
49 TK_ESPECIAL [{]
50 TK_IDENTIFICADOR [putc]
50 TK_ESPECIAL [(]
50 TK_IDENTIFICADOR [sample]
50 TK_ESPECIAL [[]
50 TK_IDENTIFICADOR [i]
50 TK_ESPECIAL []]
50 TK_ESPECIAL [)]
50 TK_ESPECIAL [;]
51 TK_IDENTIFICADOR [i]
51 TK_ESPECIAL [+]
51 TK_ESPECIAL [+]
51 TK_ESPECIAL [;]
52 TK_ESPECIAL [}]
54 TK_PR_RETURN [return]
54 TK_LIT_INT [0]
54 TK_ESPECIAL [;]
55 TK_ESPECIAL [}]
57 TK_PR_PUBLIC [public]
57 TK_PR_INT [int]
57 TK_IDENTIFICADOR [for_test]
57 TK_ESPECIAL [(]
57 TK_ESPECIAL [)]
57 TK_ESPECIAL [{]
59 TK_PR_FOR [for]
59 TK_ESPECIAL [(]
59 TK_PR_INT [int]
59 TK_IDENTIFICADOR [i]
59 TK_ESPECIAL [=]
59 TK_LIT_INT [0]
59 TK_ESPECIAL [;]
59 TK_IDENTIFICADOR [i]
59 TK_ESPECIAL [<]
59 TK_IDENTIFICADOR [strlen]
59 TK_ESPECIAL [(]
59 TK_IDENTIFICADOR [teste]
59 TK_ESPECIAL [)]
59 TK_ESPECIAL [;]
59 TK_IDENTIFICADOR [i]
59 TK_ESPECIAL [+]
59 TK_ESPECIAL [+]
59 TK_ESPECIAL [)]
59 TK_ESPECIAL [{]
60 TK_IDENTIFICADOR [putc]
60 TK_ESPECIAL [(]
60 TK_IDENTIFICADOR [sample]
60 TK_ESPECIAL [[]
60 TK_IDENTIFICADOR [i]
60 TK_ESPECIAL []]
60 TK_ESPECIAL [)]
60 TK_ESPECIAL [;]
61 TK_IDENTIFICADOR [i]
61 TK_ESPECIAL [+]
61 TK_ESPECIAL [+]
61 TK_ESPECIAL [;]
62 TK_ESPECIAL [}]
64 TK_PR_RETURN [return]
64 TK_LIT_INT [0]
64 TK_ESPECIAL [;]
65 TK_ESPECIAL [}]
67 TK_PR_PUBLIC [public]
67 TK_PR_INT [int]
67 TK_IDENTIFICADOR [do_while_test_with_if]
67 TK_ESPECIAL [(]
67 TK_ESPECIAL [)]
67 TK_ESPECIAL [{]
69 TK_PR_INT [int]
69 TK_IDENTIFICADOR [i]
69 TK_ESPECIAL [=]
69 TK_LIT_INT [0]
69 TK_ESPECIAL [;]
71 TK_PR_DO [do]
71 TK_ESPECIAL [{]
72 TK_PR_IF [if]
72 TK_ESPECIAL [(]
72 TK_IDENTIFICADOR [i]
72 TK_ESPECIAL [%]
72 TK_LIT_INT [2]
72 TK_OC_EQ [==]
72 TK_LIT_INT [0]
72 TK_ESPECIAL [)]
73 TK_IDENTIFICADOR [putc]
73 TK_ESPECIAL [(]
73 TK_IDENTIFICADOR [sample]
73 TK_ESPECIAL [[]
73 TK_IDENTIFICADOR [i]
73 TK_ESPECIAL []]
73 TK_ESPECIAL [)]
73 TK_ESPECIAL [;]
74 TK_IDENTIFICADOR [i]
74 TK_ESPECIAL [+]
74 TK_ESPECIAL [+]
74 TK_ESPECIAL [;]
75 TK_ESPECIAL [}]
75 TK_PR_WHILE [while]
75 TK_ESPECIAL [(]
75 TK_IDENTIFICADOR [i]
75 TK_ESPECIAL [<]
75 TK_IDENTIFICADOR [strlen]
75 TK_ESPECIAL [(]
75 TK_IDENTIFICADOR [teste]
75 TK_ESPECIAL [)]
75 TK_ESPECIAL [)]
75 TK_ESPECIAL [;]
77 TK_PR_RETURN [return]
77 TK_LIT_INT [0]
77 TK_ESPECIAL [;]
78 TK_ESPECIAL [}]
80 TK_PR_PUBLIC [public]
80 TK_PR_INT [int]
80 TK_IDENTIFICADOR [while_test_with_if]
80 TK_ESPECIAL [(]
80 TK_ESPECIAL [)]
80 TK_ESPECIAL [{]
82 TK_PR_INT [int]
82 TK_IDENTIFICADOR [i]
82 TK_ESPECIAL [=]
82 TK_LIT_INT [0]
82 TK_ESPECIAL [;]
84 TK_PR_WHILE [while]
84 TK_ESPECIAL [(]
84 TK_IDENTIFICADOR [i]
84 TK_ESPECIAL [<]
84 TK_IDENTIFICADOR [strlen]
84 TK_ESPECIAL [(]
84 TK_IDENTIFICADOR [teste]
84 TK_ESPECIAL [)]
84 TK_ESPECIAL [)]
84 TK_ESPECIAL [{]
85 TK_PR_IF [if]
85 TK_ESPECIAL [(]
85 TK_IDENTIFICADOR [i]
85 TK_OC_LE [<=]
85 TK_LIT_INT [5]
85 TK_ESPECIAL [)]
85 TK_ESPECIAL [{]
86 TK_IDENTIFICADOR [putc]
86 TK_ESPECIAL [(]
86 TK_IDENTIFICADOR [sample]
86 TK_ESPECIAL [[]
86 TK_IDENTIFICADOR [i]
86 TK_ESPECIAL []]
86 TK_ESPECIAL [)]
86 TK_ESPECIAL [;]
87 TK_ESPECIAL [}]
87 TK_PR_ELSE [else]
87 TK_ESPECIAL [{]
88 TK_IDENTIFICADOR [putc]
88 TK_ESPECIAL [(]
88 TK_IDENTIFICADOR [toupper]
88 TK_ESPECIAL [(]
88 TK_IDENTIFICADOR [sample]
88 TK_ESPECIAL [[]
88 TK_IDENTIFICADOR [i]
88 TK_ESPECIAL []]
88 TK_ESPECIAL [)]
88 TK_ESPECIAL [)]
89 TK_ESPECIAL [}]
90 TK_IDENTIFICADOR [i]
90 TK_ESPECIAL [+]
90 TK_ESPECIAL [+]
90 TK_ESPECIAL [;]
91 TK_ESPECIAL [}]
93 TK_PR_RETURN [return]
93 TK_LIT_INT [0]
93 TK_ESPECIAL [;]
94 TK_ESPECIAL [}]
96 TK_PR_PUBLIC [public]
96 TK_PR_INT [int]
96 TK_IDENTIFICADOR [for_test_with_if]
96 TK_ESPECIAL [(]
96 TK_ESPECIAL [)]
96 TK_ESPECIAL [{]
98 TK_PR_FOR [for]
98 TK_ESPECIAL [(]
98 TK_PR_INT [int]
98 TK_IDENTIFICADOR [i]
98 TK_ESPECIAL [=]
98 TK_LIT_INT [0]
98 TK_ESPECIAL [;]
98 TK_IDENTIFICADOR [i]
98 TK_ESPECIAL [<]
98 TK_IDENTIFICADOR [strlen]
98 TK_ESPECIAL [(]
98 TK_IDENTIFICADOR [teste]
98 TK_ESPECIAL [)]
98 TK_ESPECIAL [;]
98 TK_IDENTIFICADOR [i]
98 TK_ESPECIAL [+]
98 TK_ESPECIAL [+]
98 TK_ESPECIAL [)]
98 TK_ESPECIAL [{]
99 TK_PR_IF [if]
99 TK_ESPECIAL [(]
99 TK_IDENTIFICADOR [i]
99 TK_ESPECIAL [%]
99 TK_LIT_INT [3]
99 TK_OC_EQ [==]
99 TK_LIT_INT [0]
99 TK_ESPECIAL [)]
100 TK_PR_CONTINUE [continue]
100 TK_ESPECIAL [;]
101 TK_PR_IF [if]
101 TK_ESPECIAL [(]
101 TK_IDENTIFICADOR [i]
101 TK_ESPECIAL [%]
101 TK_LIT_INT [11]
101 TK_OC_EQ [==]
101 TK_LIT_INT [0]
101 TK_ESPECIAL [)]
102 TK_PR_BREAK [break]
102 TK_ESPECIAL [;]
103 TK_IDENTIFICADOR [putc]
103 TK_ESPECIAL [(]
103 TK_IDENTIFICADOR [sample]
103 TK_ESPECIAL [[]
103 TK_IDENTIFICADOR [i]
103 TK_ESPECIAL []]
103 TK_ESPECIAL [)]
103 TK_ESPECIAL [;]
104 TK_IDENTIFICADOR [i]
104 TK_ESPECIAL [+]
104 TK_ESPECIAL [+]
104 TK_ESPECIAL [;]
105 TK_ESPECIAL [}]
107 TK_PR_RETURN [return]
107 TK_LIT_INT [0]
107 TK_ESPECIAL [;]
108 TK_ESPECIAL [}]
109 TK_ESPECIAL [}]
